pkg/alertmanager/config.go

```diff
@@ -8,33 +8,43 @@ package alertmanager
 import (
 	"encoding/json"
 	"fmt"
+	"net/url"
 
 	"github.com/grafana/alerting/definition"
 
 	"github.com/grafana/mimir/pkg/alertmanager/alertspb"
 )
 
-// createUsableGrafanaConfig creates an AlertConfigDesc from a GrafanaAlertConfigDesc.
+// createUsableGrafanaConfig creates an amConfig from a GrafanaAlertConfigDesc.
 // If provided, it assigns the global section from the Mimir config to the Grafana config.
 // The SMTP and HTTP settings in this section can be used to configure Grafana receivers.
-func createUsableGrafanaConfig(gCfg alertspb.GrafanaAlertConfigDesc, mCfg *alertspb.AlertConfigDesc) (alertspb.AlertConfigDesc, error) {
+func createUsableGrafanaConfig(gCfg alertspb.GrafanaAlertConfigDesc, mCfg *alertspb.AlertConfigDesc) (amConfig, error) {
```

Comment:
Every time we call `createUsableGrafanaConfig` we end up using the resulting `AlertConfigDesc` within an `amConfig` struct. Creating the final `amConfig` here makes sense and reduces the duplicated lines in the calling function.